include/swift/SIL/SILConstants.h

```diff
@@ -177,6 +177,8 @@ class SymbolicValue {
   } auxInfo;
 
 public:
+  using Allocator = std::function<void *(unsigned long, unsigned)>;
```

Comment:
It would be good to document what the first and second parameters mean.

Reply:
Also: could you get away with llvm::function_ref here rather than std::function? Given that the bump pointer allocator is stack allocated in  EmitDFDiagnostics::run() , using the allocator anywhere after that will be a problem already -- so we shouldn't pay the price of std::function.